# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/public/experimental/JSStencil.h
# Commit: a10921d560a0
# Full Hash: a10921d560a0c2c178f98aefcd43207af8d95097
# Author: Tooru Fujisawa <arai_a@mac.com>
# Date: 2025-09-25 10:57 +0000
# Regressor Bug: 1980152
# File Overlap Count: 1
# Overlapping Files: dom/script/ScriptLoader.cpp
# Description:
#   Bug 1980152 - Part 1: Calculate the disk caching plan also with stencil-based requests. r=nbp
# ==============================================================================

diff -r 7adcef30c668 -r a10921d560a0 js/public/experimental/JSStencil.h
--- a/js/public/experimental/JSStencil.h	Thu Sep 25 10:57:33 2025 +0000
+++ b/js/public/experimental/JSStencil.h	Thu Sep 25 10:57:33 2025 +0000
@@ -257,6 +257,14 @@
 // This returns false if the stencil contains asm.js.
 extern JS_PUBLIC_API bool IsStencilCacheable(JS::Stencil* stencil);
 
+// ************************************************************************
+//   Script Source
+// ************************************************************************
+
+// Returns the uncompressed source text length of given stencil.
+// Returns 0 if the source is discarded or somehow not accessible.
+extern JS_PUBLIC_API size_t GetScriptSourceLength(JS::Stencil* stencil);
+
 }  // namespace JS
 
 #endif  // js_experimental_JSStencil_h